### DISCUSSION TITLE
Regarding Upgradation

### DISCUSSION THREAD

**Discussion by bagu05** [cattr-app/server-application#296] [General]
We deployed the Cattr server application two months ago in our local environment. However, there have been many updates to the source code since then. How can we upgrade it on the server?

**Comment by DanilShapilov** [comment]
Depends on how it was installed.  
1. Create a backup.
2. If installed through docker, just change docker image in docker-compose. As of right now, latest tag is :v4.0.0-RC53 https://git.amazingcat.net/cattr/core/app/container_registry/9?orderBy=NAME&sort=desc

**Reply by bagu05** [reply]
Thank you for the response.

We are not using Docker. How can we upgrade it manually?

**Reply by DanilShapilov** [reply]
Something like this.
1. Create a backup
2. I believe it should be as easy as swithcing your code to latest tag from https://github.com/cattr-app/server-application/tags using git `git checkout v4.0.0-RC53`.
3. See file `.env.example` it may have something new, I would recommend to create a new `.env` from `.env.example`  and copy your variables to it. You definately need `APP_KEY`
4. Upgrade dependencies for php via `composer install` and `yarn` for js.
5. Before running migrations, I would recommend to truncate failed_jobs table in your Cattr database.
6. Run migrations `php artisan migrate`

also build frontend, maybe clear cache in browser and in laravel
